pkg/quota/admission/runonceduration/admission.go

```diff
@@ -0,0 +1,131 @@
+package runonceduration
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+)
+
+func init() {
+	admission.RegisterPlugin("RunOnceDuration", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewRunOnceDuration(pluginConfig), nil
+	})
+}
+
+func readConfig(reader io.Reader) (*RunOnceDurationConfig, error) {
+	config := &RunOnceDurationConfig{}
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
```

Comment:
An interface may be non-nil but the pointer that it holds may be nil.
https://groups.google.com/forum/#!topic/golang-nuts/QzVDKv7p0Vs
I found that it was the case with the admission control RegisterPlugin callback that the io.Reader that was passed to it was non-nil while pointing to a nil.